pkg/ccl/crosscluster/physical/standby_read_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package physical
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/crosscluster/replicationtestutils"
+	"github.com/cockroachdb/cockroach/pkg/ccl/crosscluster/replicationutils"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/testutils/jobutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStandbyRead(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderDuress(t, "slow test")
```

Comment:
Unless you specifically are wanting to observe the behavior of the PCR job -- that is, copying bytes from one cluster to another and publishing progress updates reflecting what is copied -- I probably wouldn't use a PCR job in this test since that means you have replicate millions of rows and sit around in the associated slow WaitUntilReplicatedTime calls.

As far as I can tell, what you are intending to test here is the behavior of external-row-data query execution that fetches from a second tenant tenant, right? I'd probably suggest a simpler, faster test that just runs one cluster that has two tenants, one that you write to and one that reads from that first one as external data:

Concretely, in a test (somewhere in the sql tree?) I'd:
* create tenant twice
* write to the first tenant
* call `sql/catalog/replication.SetupOrAdvanceStandbyReaderCatalog` on the second (passing now())
* run your query on conn to the second

This way there is no PCR job, and you'll just directly move the read-as-of ts up after each write you want to observe, there's no waiting for a job to notice the change, copy it, then update progress.

Reply:
Good point, I've rewritten the test to use two tenants in a TestCluster. It now takes about 12s instead of > 60.

Reply:
Nice! Looks great. Foundations might have some knobs that speed up the lease waits to get cut that down even more.